frontend/packages/console-app/src/components/pdb/PodDisruptionBudgetField.tsx

```diff
@@ -23,24 +23,29 @@ export const PodDisruptionBudgetField: React.FC<PodDisruptionBudgetFieldProps> =
   });
   const pdb = getPDBResource(pdbResources, obj);
 
-  return (
-    <DetailsItem label={t('console-app~PodDisruptionBudgets')} obj={obj}>
-      {!_.isEmpty(pdb) ? (
-        <>
-          <ResourceLink
-            kind={referenceForModel(PodDisruptionBudgetModel)}
-            name={pdb.metadata.name}
-            namespace={pdb.metadata.namespace}
-          />
-          {obj.spec?.replicas &&
-            (!_.isNil(pdb.spec?.minAvailable) || !_.isNil(pdb.spec?.maxUnavailable)) && (
-              <AvailabilityRequirement pdb={pdb} replicas={obj.spec?.replicas} />
-            )}
-        </>
-      ) : (
-        t('console-app~No PodDisruptionBudgets')
-      )}
+  return !_.isEmpty(pdb) ? (
+    <DetailsItem
+      label={t('console-app~PodDisruptionBudgets')}
+      obj={pdb}
+      path={!_.isNil(pdb.spec?.minAvailable) ? 'spec.minAvailable' : 'spec.maxUnavailable'}
+    >
+      <ResourceLink
+        kind={referenceForModel(PodDisruptionBudgetModel)}
+        name={pdb.metadata.name}
+        namespace={pdb.metadata.namespace}
+      />
+      {obj.spec?.replicas &&
+        (!_.isNil(pdb.spec?.minAvailable) || !_.isNil(pdb.spec?.maxUnavailable)) && (
+          <AvailabilityRequirement pdb={pdb} replicas={obj.spec?.replicas} />
+        )}
     </DetailsItem>
+  ) : (
+    <DetailsItem
+      label={t('console-app~PodDisruptionBudgets')}
+      obj={pdb}
+      defaultValue={t('console-app~No PodDisruptionBudgets')}
+      description="PodDisruptionBudget is an object to define the max disruption that can be caused to a collection of pods"
+    />
   );
 };
```

Comment:
I think the DetailsItem component handles an undefined `obj` prop well as long as you don't pass in a path. The other option is that we just make the DetailsITem `obj` prop optional. In this case, we don't need it since we are passing in a custom child and description. DetailsItem only uses `obj` + `path` to get the field description and value.

Reply:
I think rather than passing in an unused `obj` prop to DetailsItem, then handling the case where it is undefined, we should change DetailsItem so that obj is an optional prop, and if it's not defined, we skip all logic depending on that prop and use fallback values that make sense. In the case of `referenceFor`, we can just use a ternary to prevent calling it when obj is undefined: `const reference = obj ? referenceFor(obj) : ''`